src/Uno.UI.Runtime.Skia.Wpf/Extensions/Storage/Pickers/FileOpenPickerExtension.cs

```diff
@@ -45,17 +45,34 @@ private async Task<StorageFile[]> OpenPickerAsync(bool multiple)
 			};
 
 			var filterBuilder = new StringBuilder();
-			filterBuilder.Append(string.Join("|", _picker.FileTypeFilter.Select(fileType => $"{fileType}|*{fileType}")));
+			var all = false;
+			var filterIndex = 0;
+			foreach (var fileType in _picker.FileTypeFilter)
+			{
+				if (fileType == "*")
```

Comment:
I think this doesn't match UWP behavior when multiple filter types are all `*`

But it's probably not important.

Reply:
This is  UWP behavior
![immagine](https://user-images.githubusercontent.com/12531229/146911981-d7b19ddf-0cd2-4f4c-a5ec-65f6a58f9eda.png)

Reply:
@workgroupengineering I mean when having multiple `*` specified. This is how UWP behaves:

![image](https://user-images.githubusercontent.com/31348972/146941362-c444aa0e-d169-4270-bf1b-23ebdeeb0275.png)

Reply:
I think it's more `*.*` on Windows for _all files_, no?  Here's a screenshot from my NotePad:
![image](https://user-images.githubusercontent.com/4174207/146974796-5b9b59e3-030b-4177-9363-ed624703b400.png)

Reply:
@carldebilly This is not how it behaves on UWP I think.

Reply:
> @workgroupengineering I mean when having multiple `*` specified. This is how UWP behaves:
> 
> ![image](https://user-images.githubusercontent.com/31348972/146941362-c444aa0e-d169-4270-bf1b-23ebdeeb0275.png)
> 
> It's meaningless to specify `*` multiple times, that's why it's not much important.

It sounds strange, but in UWP the behavior is the same, I have checked now.



> @carldebilly This is not how it behaves on UWP I think.
I confirm

Reply:
> It sounds strange, but in UWP the behavior is the same, I have checked now.

Yup with the latest commits this got fixed. The initial commit when I reviewed didn't match UWP for this weird edge case.